libass/ass_parse.c

```diff
@@ -501,14 +501,13 @@ char *parse_tag(ASS_Renderer *render_priv, char *p, char *end, double pwr)
             t1 = 0;
             t2 = render_priv->state.event->Duration;
         }
-        delta_t = t2 - t1;
         t = render_priv->time - render_priv->state.event->Start;
         if (t <= t1)
             k = 0.;
         else if (t >= t2)
             k = 1.;
         else
-            k = ((double) (t - t1)) / delta_t;
+            k = subtract(t, t1) / subtract(t2, t1);
```

Comment:
That's essentially continuation of the previous commit, so I think that should be squashed too.